src/common/atlas/apiClient.ts

```
@@ -48,32 +39,51 @@ export interface ApiClientOptions {
 }
 
 export class ApiClient {
-    token?: OAuthToken;
-    saveToken?: saveTokenFunction;
+    private token?: OAuthToken;
+    private saveToken?: saveTokenFunction;
+    private client = createClient<paths>({
+        baseUrl: config.apiBaseUrl,
+        headers: {
+            "User-Agent": config.userAgent,
+            Accept: `application/vnd.atlas.${config.atlasApiVersion}+json`,
+        },
+    });
+    private authMiddleware = (apiClient: ApiClient): Middleware => ({
+        async onRequest({ request, schemaPath }) {
+            if (schemaPath.startsWith("/api/private/unauth") || schemaPath.startsWith("/api/oauth")) {
+                return undefined;
+            }
+            if (await apiClient.validateToken()) {
+                request.headers.set("Authorization", `Bearer ${apiClient.token?.access_token}`);
+                return request;
+            }
+        },
+    });
+    private errorMiddleware = (): Middleware => ({
+        async onResponse({ response }) {
+            if (!response.ok) {
+                try {
+                    const text = await response.text();
+                    throw new ApiClientError(
```

this does mean that what we'll get is 
`Error calling Atlas API: Error calling Atlas API:... `

---

I think passing the response is sufficient, we can just omit the response text

---

What do you mean? response is a property and does not display error message directly. You prefer to hide error message?

---

I'd say so. If we want to include properties of the response inside the message (including in format `[${response.status} ${response.statusText}]`) then let's do it inside the `ApiClientError` class's `super` call so it's standardized and enforced. And in that sense avoid anything async.